src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.ts

```diff
@@ -345,10 +348,21 @@ export class InstructorSessionsPageComponent extends InstructorSessionModalPageC
       });
     }, (resp: ErrorMessageOutput) => {
       this.sessionEditFormModel.isSaving = false;
-      this.statusMessageService.showErrorToast(resp.error.message);
+      this.statusMessageService.showErrorToast(
+          this.errorMessageFormatting(this.sessionEditFormModel.feedbackSessionName,
+              this.courseId, resp.error.message));
     });
   }
 
+  errorMessageFormatting(feedbackSessionName: string, courseId: string, errorMessage: string): string {
+    if (errorMessage.match('Trying to create an entity that exists')) {
```

Comment:
While this works, why not just move the message to the API response itself?

Yes, I did mention `This API has no business with "sessions page" or "recycle bin [table]"`, but the sentence before that `A session named ... already existed in the course` is a valid back-end validation message, because it correctly describes why the back-end cannot accept the request to create new session.

The only thing that the front-end needs to add further is the second sentence.

Reply:
Sorry for the misunderstanding and thanks for the prompt review! I have made some changes to the code again :)